packages/compass-home/src/modules/hashed-namespace.ts

```diff
@@ -0,0 +1,20 @@
+const HashedNamespace = new Map<string, Promise<string>>();
+
+export async function getHashedNamespace(ns: string): Promise<string> {
+  if (HashedNamespace.has(ns)) {
+    // We just checked that the property exists
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return HashedNamespace.get(ns)!;
+  }
+  const hashPromise = (async () => {
+    const buffer = await crypto.subtle.digest(
+      'SHA-256',
+      new TextEncoder().encode(ns)
+    );
```

Comment:
What is the hash intended to achieve? If the goal is to prevent us as owners of the telemetry from identifying collection names, it’s going to be doing a mediocre job at it, because database/collection names are fairly low-entropy, and it might be a good idea to include additional information (e.g. cluster hostnames, local compass user UUID)

Reply:
My thinking was that yes, just doing a one way hashing here would be enough, thanks for highlighting. Do you think adding just user id would be enough to make it non-reversible? I think that's the only value we can sort of safely get in this particular context without doing scary things with existing architecture. If not, I'll just drop namespace from the telemetry and we can create a separate project to figure out exactly how to change Compass to account for that if we ever want to do this sort of telemetry setup

Reply:
> Do you think adding just user id would be enough to make it non-reversible

I mean … if the user id is the one we’re sending with the telemetry, it doesn’t change all that much. But otherwise something like a persisted UUID is probably ideal

Reply:
Okay, sounds like we'd need to have a design for this. I'll remove namespace from telemetry

Reply:
Reverted and removed the namespace in https://github.com/mongodb-js/compass/pull/4576/commits/d7801240dee14037523a411b8b2ee605f0a4d109